Makefile

```
@@ -1,118 +1,64 @@
-# Shortcut targets
-default: build
-
-## Build binary for current platform
-all: build
-
-## Run the tests for the current platform/architecture
-test: fv st
+PACKAGE_NAME?=github.com/projectcalico/node
+GO_BUILD_VER?=v0.27
 
 ###############################################################################
-# Both native and cross architecture builds are supported.
-# The target architecture is select by setting the ARCH variable.
-# When ARCH is undefined it is set to the detected host architecture.
-# When ARCH differs from the host architecture a crossbuild will be performed.
-ARCHES=$(patsubst Dockerfile.%,%,$(wildcard Dockerfile.*))
-
-# BUILDARCH is the host architecture
-# ARCH is the target architecture
-# we need to keep track of them separately
-BUILDARCH ?= $(shell uname -m)
-BUILDOS ?= $(shell uname -s | tr A-Z a-z)
-
-# canonicalized names for host architecture
-ifeq ($(BUILDARCH),aarch64)
-	BUILDARCH=arm64
-endif
-ifeq ($(BUILDARCH),x86_64)
-	BUILDARCH=amd64
-endif
+# Download and include Makefile.common
+#   Additions to EXTRA_DOCKER_ARGS need to happen before the include since
+#   that variable is evaluated when we declare DOCKER_RUN and siblings.
+###############################################################################
+MAKE_BRANCH?=$(GO_BUILD_VER)
+MAKE_REPO?=https://raw.githubusercontent.com/projectcalico/go-build/$(MAKE_BRANCH)
+WGET?=/usr/bin/wget
 
-# unless otherwise set, I am building for my own architecture, i.e. not cross-compiling
-ARCH ?= $(BUILDARCH)
+Makefile.common: Makefile.common.$(MAKE_BRANCH)
+	cp "$<" "$@"
+Makefile.common.$(MAKE_BRANCH): $(WGET)
+	# Clean up any files downloaded from other branches so they don't accumulate.
+	rm -f Makefile.common.*
+	$(WGET) -nv $(MAKE_REPO)/Makefile.common -O "$@"
```

Why this instead of just relying on whatever is in `$PATH`?
```suggestion
	wget -nv $(MAKE_REPO)/Makefile.common -O "$@"
```

---

Gaurav asked to add it as a dependency.

---

It'd be nice if we could add it as a dependency without requiring it exist at `/usr/bin/wget`. But maybe not worth the effort because for most (all?) systems it is there. Is it there for Mac OS too?

---

Will check on where Mac OS puts it...

---

Seems that Mac OS doesn't ship wget, so it's up to the user to install it. Should I move all of the repos to using curl(1) or just rely on the user to install/link to it in the right place? I've no preference on this tbh.

---

Ok, I'm moving all repos to curl and removing the dependency with a separate set of PRs.